ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/docker_entrypoint.sh

```diff
@@ -0,0 +1,113 @@
+set -eo pipefail
+
+TARGET_SYNTHEA_DIR=$(readlink -f "$(dirname "${BASH_SOURCE[0]}")")
+readonly TARGET_SYNTHEA_DIR
+
+BFD_END_STATE_PROPERTIES="$TARGET_SYNTHEA_DIR/end_state.properties"
+readonly BFD_END_STATE_PROPERTIES
+
+NUM_REGEX="^[0-9]+$"
+readonly NUM_REGEX
+
+num_generated_benes=100
+generate_future="false"
+num_future_months=0
+
+help() {
+  echo
+  echo "Docker container for running Synthea generation:"
+  echo
+  echo "-n : Specifies the number of beneficiaries to generate. Defaults to 100"
+  echo "-f : Specifies the number of months in the future to generate claim lines. Defaults to 0"
+  echo "-v : Specifies whether the Synthea executable should log to STDOUT or if its logs should be discared. Defaults to unset (non-verbose)"
+  echo "-h : Shows this help text"
+  exit 0
+}
+
+while getopts ":n:f:vh" opt; do
+  case $opt in
+    n)
+      if ! [[ "$OPTARG" =~ $NUM_REGEX ]]; then
+        echo "ERROR, non-numeric specified ($OPTARG)...exiting" >&2
+        exit 1
+      fi
+      num_generated_benes="$OPTARG"
+      ;;
+    f)
+      if ! [[ "$OPTARG" =~ $NUM_REGEX ]]; then
+        echo "ERROR, non-numeric future month value specified ($OPTARG)...exiting" >&2
+        exit 1
+      fi
+      num_future_months="$OPTARG"
+      if [[ "$OPTARG" -gt 0 ]]; then
+        generate_future="true"
+      fi
+      ;;
+    v)
+      export VERBOSE_SYNTHEA_LOGS="true"
+      ;;
+    h)
+      help
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      ;;
+  esac
+  shift
+done
+
+echo "Checking if the output directory is empty before generation..."
+(
+  cd "$TARGET_SYNTHEA_DIR/output"
+  shopt -s nullglob
+  files=(* .*)
+  if ((${#files[@]} != 2)); then
+    # contents of files array is (. ..)
+    echo "The output directory is not empty; ensure the directory is empty before running"
+    exit 1
+  fi
+)
+
+echo "Preparing to run Synthea generation..."
+starting_datetime=$(date '+%F_%H-%M-%S')
+
+echo "Running Synthea generation with $num_generated_benes benes and $num_future_months future months..."
+{
+  python3 prepare-and-run-synthea.py \
+    "${BFD_END_STATE_PROPERTIES}" \
+    "${TARGET_SYNTHEA_DIR}" \
+    "${num_generated_benes}" \
+    "${num_future_months}" 2>&1 &&
+  echo "Synthea generation finished, generated synthetic data can be found in the output directory"
+} || {
+  echo "Synthea generation failed to complete. View the logs in the logs directory for more information"
+}
+
+if [ "$generate_future" == 'true' ]; then
+  echo "Generating future months was specified, splitting future claims to allow for proper loading..."
+  {
+    python3 split-future-claims.py "$TARGET_SYNTHEA_DIR" \
+      2>&1 | tee -a "$TARGET_SYNTHEA_DIR/logs/split_future_claims.latest.log" &&
+    echo "Future claims splitting succeeded, view the split_future_claims-$starting_datetime.log for more information"
+  } || {
+    echo "Future claims splitting failed, view the split_future_claims-$starting_datetime.log for more information"
```

Comment:
Any script output should be printing to stdout anyway, right? They should just be able to look at the jenkins log.

Reply:
Agreed. I will update this log message indicating that either `stdout` _or_ the distinct log file can be checked.

Reply:
I've updated this log message to indicate that the logs can be viewed in the "full log" (i.e. `stdout`, Jenkins log) _or_ the distinct log file.